### PR TITLE
PMP doc fixes and enhancements

### LINKED ISSUE #6385: PMP: Navtree and Reference Manual Index are not in sync
## Issue Details

If you look at "Combinatorial Repairing Functions" you can see that the entry page of the Reference Manual is not complete. 

image

### PR DESCRIPTION
**Pull request by MaelRL**
## Summary of Changes

Spring cleaning for PMP's doc.

## Release Management

- [ ] Post-merge: handle group label changes 
hole_filling_grp -> PMP_hole_filling_grp 
measure_grp -> PMP_measure_grp
keep_connected_components_grp -> PMP_keep_connected_components_grp

* Affected package(s):  `PMP`
* Issue(s) solved (if any): fix #6385
* Feature/Small Feature (if any): -
* License and copyright ownership: no change

### CODE REVIEW

**Comment by sloriot** [comment]
Successfully tested in CGAL-5.5-Ic-36

File: Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/IO/polygon_mesh_io.h [hunk 1]
```diff
@@ -37,7 +37,8 @@ namespace IO {
 /*!
   \ingroup PMP_IO_grp
 
- * \brief reads the file as a polygon soup, repairs, and orients it as to obtain a polygon mesh.
+ * \brief reads the file as a polygon soup, repairs (using `repair_polygon_soup()`),
+ * and orients it (using `orient_polygon_soup()`) as to obtain a polygon mesh.
```
Comment (afabri) [inline Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/IO/polygon_mesh_io.h:41]: I would still suggest to add a sentence like     "If there is no need to orient and repair the input you should use the function   `CGAL::IO::read_polygon_mesh()`"

**Review by afabri — CHANGES_REQUESTED** [review]

**Comment by sloriot** [comment]
Successfully tested in CGAL-5.5-Ic-50

**Pull request merged**